src/modding/FullModDetails.cs

```diff
@@ -1,22 +1,54 @@
 ﻿using System;
+using System.Collections.Generic;
+using Godot;
+using Newtonsoft.Json;
 
 /// <summary>
 ///   An extended version of <see cref="ModInfo"/> that contains non-mod author controlled data
 /// </summary>
 public class FullModDetails : IEquatable<FullModDetails>
 {
+    [JsonConstructor]
     public FullModDetails(string internalName, string folder, ModInfo info)
     {
         InternalName = internalName;
         Folder = folder;
         Info = info;
     }
 
+    public FullModDetails(string internalName)
+    {
+        InternalName = internalName;
+        Info = new ModInfo();
+        Folder = string.Empty;
+    }
+
     public string InternalName { get; }
     public string Folder { get; set; }
 
+    /// <summary>
+    ///   Is the mod compatible with the current version of thrive?
+    /// </summary>
+
+    // 1 = Compatible, 0 = Unknown, -1 = Might Not Be Compatible, -2 = Not Compatible
+    public int IsCompatibleVersion { get; set; }
```

Comment:
This is very weird a doc. You should probably add the comment line as a `<remarks>` (with an inner `<para>`, as per styleguide).

Reply:
Having to document specific values also seems like this maybe should be an enum value.
In case we don't have proper deserialization from enum value name to enum yet, there is one converter I wrote for the devcenter: https://github.com/Revolutionary-Games/ThriveDevCenter/blob/master/Shared/Converters/ActualEnumStringConverter.cs